test/python/pulse/test_schedule.py

```diff
@@ -463,16 +477,30 @@ def test_filter_inst_types(self):
         sched = sched.insert(60, acquire(device.acquires, device.memoryslots))
         sched = sched.insert(90, lp0(device.drives[0]))
 
-        only_acquires = sched.filter(instruction_types=[AcquireInstruction])
-        for _, inst in only_acquires.instructions:
+        # split into parts with and without AcquireInstruction
```

Comment:
The code that you've written here is self explanatory. (yay!) `only_acquire, no_acquire = self.filter(schedule, ...)` tells me that I am splitting the schedule into pieces with acquires and pieces without. When code is clear, we don't need to repeat the same info in a comment. Some people do love having superfluous comments, so this is just one way to think about it. Usually, when I write a comment, I ask myself, is there a way to rewrite my code that this comment wouldn't be necessary? Here, this test is doing a lot. We are using comments mainly as headers to say "test on Acquire", "now test of two other instruction types", "lastly just the framechange". We could shorten the comments to highlight that they _are_ headers and just write the instruction types we're testing, or we could make this into three separate tests.